### PR TITLE
max length extension

### PR DESCRIPTION
**Pull request by enriquesouza**
https://github.com/vultisig/vultisig-ios/issues/617

### CODE REVIEW

File: VultisigApp/VultisigApp/Extensions/TextField.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import SwiftUI
+import Combine
+
+#if canImport(UIKit)
+import UIKit
+
+private var maxLengthKey: UInt8 = 0
+
+extension UITextField {
+    @IBInspectable var maxLength: Int {
```
Comment (flypaper0) [inline VultisigApp/VultisigApp/Extensions/TextField.swift:19]: IBInspectable can be removed, we're do not use interface builder

File: VultisigApp/VultisigApp/Extensions/TextField.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        get {
+            return objc_getAssociatedObject(self, &maxLengthKey) as? Int ?? 50
```
Comment (flypaper0) [inline VultisigApp/VultisigApp/Extensions/TextField.swift:21]: why 50 is default value?

File: VultisigApp/VultisigApp/Extensions/TextField.swift [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import Foundation
+import SwiftUI
+import Combine
+
+#if canImport(UIKit)
+import UIKit
+
+private var maxLengthKey: UInt8 = 0
+
+extension UITextField {
```
Comment (flypaper0) [inline VultisigApp/VultisigApp/Extensions/TextField.swift:18]: This extension is unused. Let's remove as we're not using UITextFields from UIKit

**Review by flypaper0 — APPROVED** [review]

File: VultisigApp/VultisigApp/Extensions/TextField.swift [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+}
+#endif
+
+struct MaxLengthModifier: ViewModifier {
+    @Binding var text: String
+    var maxLength: Int = 50
+    
+    func body(content: Content) -> some View {
+        content
+            .onChange(of: text) { newValue in
```
Comment (flypaper0) [inline VultisigApp/VultisigApp/Extensions/TextField.swift:44]: Fix warning please 
`'onChange(of:perform:)' was deprecated in iOS 17.0: Use `onChange` with a two or zero parameter action closure instead.`

File: VultisigApp/VultisigApp/Extensions/TextField.swift (hunk 3 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Extensions/TextField.swift:18]: It is used to set max length to both macOs and IOs.

File: VultisigApp/VultisigApp/Extensions/TextField.swift (hunk 4 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Extensions/TextField.swift:44]: Ah, sure

File: VultisigApp/VultisigApp/Extensions/TextField.swift (hunk 1 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Extensions/TextField.swift:19]: It is used to set max length to both macOs and IOs.

Comment (johnnyluo) [reply VultisigApp/VultisigApp/Extensions/TextField.swift:19]: I think what @flypaper0 means is `@IBInspectable` is used for interface builder , we are not using that , so the property wrapper should be removed

Comment (enriquesouza) [reply VultisigApp/VultisigApp/Extensions/TextField.swift:19]: Hum... let me check

File: VultisigApp/VultisigApp/Extensions/TextField.swift (hunk 3 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Extensions/TextField.swift]: You are right.

File: VultisigApp/VultisigApp/Extensions/TextField.swift (hunk 1 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Extensions/TextField.swift]: He is right, we don't need that the extension resolves that. Thanks @flypaper0

**Pull request merged**